libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt

```diff
@@ -0,0 +1,157 @@
+ * @param names the edge's names
+ * @param curvature the edge's curvature
+ * @param geometry optional geometry if requested
+ * @param speedLimit the Edge's max speed
+ * @param laneCount the edge's lane counts
+ * @param meanElevation the edge's mean elevation
+ * @param countryCode the edge's country code
+ * @param stateCode the edge's state code
+ */
+class Edge internal constructor(
+    val id: Long,
+    val level: Byte,
+    val probability: Double,
+    val heading: Double,
+    val length: Double,
+    val out: List<Edge>,
+    var parent: Edge?,
+    val frc: String,
+    val speed: Double,
+    val ramp: Boolean,
+    val motorway: Boolean,
+    val bridge: Boolean,
+    val tunnel: Boolean,
+    val toll: Boolean,
+    val names: List<NameInfo>,
+    val curvature: Byte,
+    val geometry: LineString?,
+    val speedLimit: Double?,
+    val laneCount: Byte?,
+    val meanElevation: Double?,
+    val countryCode: String?,
+    val stateCode: String?
+) {
+
+    /**
+     * @return true if the Edge is the MPP, false if not
+     */
+    fun isMpp(): Boolean {
+        return level == 0.toByte()
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Edge
+
+        if (id != other.id) return false
+        if (level != other.level) return false
+        if (probability != other.probability) return false
+        if (heading != other.heading) return false
+        if (length != other.length) return false
+        if (out != other.out) return false
+        if (parent != other.parent) return false
+        if (frc != other.frc) return false
+        if (speed != other.speed) return false
+        if (ramp != other.ramp) return false
+        if (motorway != other.motorway) return false
+        if (bridge != other.bridge) return false
+        if (tunnel != other.tunnel) return false
+        if (toll != other.toll) return false
+        if (names != other.names) return false
+        if (curvature != other.curvature) return false
+        if (geometry != other.geometry) return false
+        if (speedLimit != other.speedLimit) return false
+        if (laneCount != other.laneCount) return false
+        if (meanElevation != other.meanElevation) return false
+        if (countryCode != other.countryCode) return false
+        if (stateCode != other.stateCode) return false
+
+        return true
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun hashCode(): Int {
+        var result = id.hashCode()
+        result = 31 * result + level.hashCode()
+        result = 31 * result + probability.hashCode()
+        result = 31 * result + heading.hashCode()
+        result = 31 * result + length.hashCode()
+        result = 31 * result + out.hashCode()
+        result = 31 * result + parent.hashCode()
```

Comment:
this would have a stack overflow too right? 🤔

Reply:
Not sure. It seems `hashCode` is different. Didn't run into any while testing 🤔